### PR TITLE
chore: remove npm deps from dependabot + auto-approve

### PR DESCRIPTION
**Pull request by eladb**
We have a custom workflow which updates deps.

Additionally, add a GitHub workflow to auto-approve PRs with the `pr/auto-approve` label.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: .github/workflows/auto-approve.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+on: pull_request
+
+jobs:
+  auto-approve:
+    if: contains(github.event.pull_request.labels.*.name, 'pr/auto-approve')
+    runs-on: ubuntu-latest
+    steps:
+    - uses: hmarr/auto-approve-action@v2.0.0
+      with:
+        github-token: "${{ secrets.AUTO_APPROVE_GITHUB_TOKEN }}"
```
Comment (RomainMuller) [inline .github/workflows/auto-approve.yml:13]: We already have such a workflow, which does not use `pr/auto-approve`, and instead triggers on the author...

https://github.com/aws/jsii/blob/716a8994dc46cc44d38ae61289a2193b3b16ebec/.github/workflows/dependabot.yml#L1-L28

Why create yet another one?

**Review by RomainMuller — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-approve.yml (hunk 1 above)
Comment (eladb) [reply .github/workflows/auto-approve.yml:13]: The `pr/auto-approve` is more generic and can be applied to additional automation tasks. This is what we are normalizing on for the CDK repo.

Comment (eladb) [reply .github/workflows/auto-approve.yml:13]: Deleted the old one

Comment (RomainMuller) [reply .github/workflows/auto-approve.yml:13]: Well sure but you are giving away one property of the old workflow with __YOU__ had insisted be added: the ability to only auto-approve changes that have an impact of "minor" or less (i.e: the dependency is verified not to be a major version bump).

Comment (eladb) [reply .github/workflows/auto-approve.yml:13]: Not sure I understand. Isn't this taken care of by the yarn-update workflow?

**Review by RomainMuller — APPROVED** [review]

File: .github/workflows/auto-approve.yml (hunk 1 above)
Comment (eladb) [reply .github/workflows/auto-approve.yml:13]: okay, got it. this workflow only auto-approve dependabot PRs for minor versions and not major versions.

**Pull request merged**